sky/provision/kubernetes/instance.py

```diff
@@ -482,17 +481,19 @@ def _create_pods(region: str, cluster_name_on_cloud: str,
     created_pods = {}
     logger.debug(f'run_instances: calling create_namespaced_pod '
                  f'(count={to_start_count}).')
-    for _ in range(to_start_count):
+    for pod_id in range(config.count):
         if head_pod_name is None:
             pod_spec['metadata']['labels'].update(constants.HEAD_NODE_TAGS)
             head_selector = head_service_selector(cluster_name_on_cloud)
             pod_spec['metadata']['labels'].update(head_selector)
             pod_spec['metadata']['name'] = f'{cluster_name_on_cloud}-head'
         else:
             pod_spec['metadata']['labels'].update(constants.WORKER_NODE_TAGS)
-            pod_uuid = str(uuid.uuid4())[:4]
-            pod_name = f'{cluster_name_on_cloud}-{pod_uuid}'
-            pod_spec['metadata']['name'] = f'{pod_name}-worker'
+            pod_name = f'{cluster_name_on_cloud}-worker{pod_id}'
```

Comment:
One of the reasons we used uuid is because the SSH config entries do not directly map to worker id. E.g., on this PR, I ran `sky launch -c test2 --cloud kubernetes --num-nodes 3 -- echo hi` followed by `ssh test2-worker1`, and I was ssh'd into test2-worker2 pod.

```
$ ssh test2-worker1
# Logged into pod named worker2
(base) sky@test2-2ea4-worker2:~$
```

Perhaps we need some handling in how SSH configs are generated?

Reply:
ok so looks what's happening is when the ssh config is being generated, the ips are sorted [here](https://github.com/skypilot-org/skypilot/blob/1c32aa4ad349e4dc1f38e2e24537bc0b6ebfdae9/sky/backends/cloud_vm_ray_backend.py#L2338). So if two pods are created with the following mapping
```
worker1 10.44.3.19
worker2 10.44.1.20
```
the sort incorrectly flips the mapping which also explains why it's non-deterministic.
```
worker1 10.44.1.20
worker2 10.44.3.19
```

Question regarding `update_cluster_ips` in `cloud_vm_ray_backend.py` for `kubernetes`? Is it always going to use `internal_ips`? I imagine there's no external ips associated with pods. Just wondering if it's possible to just skip the sort step altogether as the logic seems tied to using external ips.

Reply:
fixed by removing the sorting logic in `update_cluster_ips`. resolving